functions/go/set-project-id/README.md

```diff
@@ -0,0 +1,79 @@
+# set-project-id
+
+## Overview
+
+<!--mdtogo:Short-->
+
+The `set-project-id` function sets 'project-id' setter and
+`cnrm.cloud.google.com/project-id` annotation to the provided project ID, only
+if they are not already set.
+
+<!--mdtogo-->
+
+<!--mdtogo:Long-->
+
+## Usage
+
+list-setters function is expected to be executed imperatively like:
+
+```shell
+kpt fn eval --include-meta-resources --image gcr.io/kpt-fn/set-project-id:v0.1 -- 'project-id=foo'
+```
+
+The `set-project-id` function does the following:
+
+1.  Sets the the 'project-id' setter to the provided project ID.
+    *   If an 'apply-setters' function is present in the Kptfile which does not
+        have the 'project-id' setter or sets it to an empty value, update that
+        config to set the “project-id” setter to the provided project ID value.
+    *   If no 'apply-setters' function is present in the Kptfile, add one with
```

Comment:
What if there is not pipeline declare in the Kptfile?

Reply:
It will declare a pipeline and add 'apply-setters' mutator with 'project-id' setter. Added this to the README.